### DISCUSSION TITLE
Is there way to add new entity type?

### DISCUSSION THREAD

**Discussion by 42zeroo** [saleor/saleor#9392] [Q&A]
Hi saleor team!
I'm planning to build new ecommerce site on saleor engine.
Constructing structure of products, categories, taxes etc looks pretty well and as for now all tests cover requirements.
But my client wants to has easy way to controll all content on page.
If client will want to create simple pages - easy to do by adding new ``page_type`` with specific fields.
But problem is when client wants to be able to menage every section on homepage, by choosing one he like (from select list i think), fill content fields conditionally showed and menage order of sections. 
But to cover this feature(which i cannot found) (i think) I need to add new entity called "sections", and after that im wondering to have ability to add new "content_type" of newly created ``section`` entity. Then i want to create couple of section types, then create just final sections. Then i have idea to render them on ``Page`` by reference attribute. Is it a good way of attempting to this system? And is it also possible to do simple(or not)? I cant find way in system or even in docs how to add new entity type. If not, is best way to do it by create module by self?

Kind regards!

**Comment by krzysztofwolski** [comment]
Hello! 
An alternative approach could be using menus instead of pages:
https://github.com/saleor/react-storefront/blob/main/pages/%5Bchannel%5D/%5Blocale%5D/index.tsx

You'll be able to use page references or collections/categories.
